test/Kestrel.FunctionalTests/Http2/H2SpecTests.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETCOREAPP2_2
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2
+{
+    [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "Missing SslStream ALPN support: https://github.com/dotnet/corefx/issues/30492")]
+    [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win81,
+        SkipReason = "Missing Windows ALPN support: https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation#Support")]
```

Comment:
Could we add a condition based on being able to execute h2spec just in case we ever try to run tests on alpine or something like that?

Reply:
You don't expect H2spec to be able to run on alpine? 

Actually, if we can't run h2spec then the test enumeration will fail and we'll never get to the skip conditions. This will fail the test run. That seems preferable until we at least know why and where it fails.

Reply:
Will the other tests still run? If so, that seems fine. Alpine uses musl. According to the [musl faq](https://www.musl-libc.org/faq.html)

> At present, some glibc-linked shared libraries can be loaded with musl, but all but the simplest glibc-linked applications will fail if musl is dropped-in in place of /lib/ld-linux.so.2.

I know our libuv binaries segfault when used on Alpine, and that's a shared library.

Reply:
Yes I'd expect the other tests to still run unless it was a catastrophic failure.